packages/polling-controller/src/PollingController.ts

```diff
@@ -1,150 +1,144 @@
-   *
-   * @param pollingToken - The polling token to stop polling for
-   */
-  stop(pollingToken: string) {
-    if (!pollingToken) {
-      throw new Error('pollingToken required');
     }
-    let found = false;
-    this.networkClientIdTokensMap.forEach((tokens, networkClientId) => {
-      if (tokens.has(pollingToken)) {
-        found = true;
-        this.networkClientIdTokensMap
-          .get(networkClientId)
-          ?.delete(pollingToken);
-        if (this.networkClientIdTokensMap.get(networkClientId)?.size === 0) {
-          clearTimeout(this.intervalIds[networkClientId]);
-          delete this.intervalIds[networkClientId];
-          this.networkClientIdTokensMap.delete(networkClientId);
-          this.messagingSystem.publish(
-            `${this.name}:pollingComplete`,
-            networkClientId,
-          );
+
+    /**
+     * Stops polling for a networkClientId
+     *
+     * @param pollingToken - The polling token to stop polling for
+     */
+    stop(pollingToken: string) {
+      if (!pollingToken) {
+        throw new Error('pollingToken required');
+      }
+      let found = false;
+      this.#networkClientIdTokensMap.forEach((tokens, networkClientId) => {
+        if (tokens.has(pollingToken)) {
+          found = true;
+          this.#networkClientIdTokensMap
+            .get(networkClientId)
+            ?.delete(pollingToken);
+          if (this.#networkClientIdTokensMap.get(networkClientId)?.size === 0) {
+            clearTimeout(this.#intervalIds[networkClientId]);
+            delete this.#intervalIds[networkClientId];
+            this.#networkClientIdTokensMap.delete(networkClientId);
+            this.#callbacks.get(networkClientId)?.forEach((callback) => {
+              callback(networkClientId);
+            });
+            this.#callbacks.get(networkClientId)?.clear();
+          }
         }
+      });
+      if (!found) {
+        throw new Error('pollingToken not found');
       }
-    });
-    if (!found) {
-      throw new Error('pollingToken not found');
     }
-  }
 
-  /**
-   * Executes the poll for a networkClientId
-   *
-   * @param networkClientId - The networkClientId to execute the poll for
-   */
-  abstract executePoll(networkClientId: NetworkClientId): Promise<void>;
+    /**
+     * Executes the poll for a networkClientId
+     *
+     * @param networkClientId - The networkClientId to execute the poll for
+     */
+    abstract executePoll(networkClientId: NetworkClientId): Promise<void>;
 
-  #poll(networkClientId: NetworkClientId) {
-    if (this.intervalIds[networkClientId]) {
-      clearTimeout(this.intervalIds[networkClientId]);
-      delete this.intervalIds[networkClientId];
+    #poll(networkClientId: NetworkClientId) {
+      if (this.#intervalIds[networkClientId]) {
+        clearTimeout(this.#intervalIds[networkClientId]);
+        delete this.#intervalIds[networkClientId];
+      }
+      this.#intervalIds[networkClientId] = setTimeout(async () => {
```

Comment:
To resolve the ESLint warning on this line:

```suggestion
      // setTimeout is not `await`ing this async function, which is expected
      // We're just using async here for improved stack traces
      // eslint-disable-next-line @typescript-eslint/no-misused-promises
      this.#intervalIds[networkClientId] = setTimeout(async () => {
```